src/states/elder/mod.rs

```diff
@@ -373,7 +373,7 @@ impl Elder {
                 }
 
                 // Drop: condition may have changed.
-                AccumulatingEvent::OurMerge => false,
+                AccumulatingEvent::OurMerge | AccumulatingEvent::Custom(_) => false,
```

Comment:
I'm not sure about this, but we might want to carry the custom events through prefix changes.

Reply:
>we might want to carry the custom events through prefix changes.

Wasn't sure about this either. Probably yes, it's better to leave the filtering to a user: so e.g. in the Vaults case we can decide if these events are still relevant to us if we know the prefix has changed.

Reply:
Should work. Assuming vaults get notified about prefix changes.